mesonbuild/compilers/c.py

```diff
@@ -353,6 +353,28 @@ def __init__(self, exelist, version, for_machine: MachineChoice,
                            info, exe_wrap, **kwargs)
         MSVCCompiler.__init__(self, target)
 
+    def get_options(self):
+        opts = super().get_options()
+        c_stds = ['none', 'c89', 'c99', 'c11']
+        opts.update({
+            'std': coredata.UserComboOption(
+                'C language standard to use',
+                c_stds,
+                'none',
+            ),
+        })
+        return opts
+
+    def get_option_compile_args(self, options):
+        args = []
+        std = options['std']
+        # As of MVSC 16.7, /std:c11 is the only valid C standard option.
+        # default is c99-like, with c11 requiring /std:c11
+        # Add more as MSVC adds C standards in the future
```

Comment:
Are you sure about this? I thought the default is C89 like, not C99 like (no designated initializers come to mind)

Reply:
Oh thanks I need to check into that thank you

Reply:
based on some [toy examples](https://github.com/scivision/c11-examples) and docs, there are a scattershot of C99 features in MSVC 19.27 as required to support C++.  For example C99 `restrict` is in MSVC 19.27 but not 19.26.